src/lumen/activity/field.cpp

```diff
@@ -0,0 +1,64 @@
+#include "lumen/activity/field.hpp"
+
+namespace lumen::activity {
+
+// Field implementation.
+template <typename T>
+void Field<T>::signal_update()
+{
+    if (parent_ != nullptr) {
+        parent_->update_display();
+    }
+}
+
+// NumberField implementation.
+NumberField::NumberField(Activity* parent, uint increment /* = 1 */)
+    : Field{parent, {}}, increment_{increment}
+{
+}
+
+NumberField& NumberField::increase()
+{
+    set_value(get_value() + increment_);
+
+    return *this;
+}
+
+NumberField& NumberField::decrease()
+{
+    auto value = get_value();
+
+    if (value >= increment_) {
+        set_value(value - increment_);
+    }
+
+    return *this;
```

Comment:
Can we just `return this;` since the return type is pass by reference?

Reply:
No, `this` is a pointer so we need to dereference it before we return it.